### PR TITLE
NOISSUE - Add channel update

### PR DESCRIPTION
**Pull request by fbugarski**
Signed-off-by: fbugarski <filipbugarski@gmail.com>

Pull request title should be `MF-XXX - description` or `NOISSUE - description` where XXX is ID of issue that this PR relate to.
Please review the [CONTRIBUTING.md](./CONTRIBUTING.md) file for detailed contributing guidelines.

### Which issue(s) does this PR fix/relate to?
Put here `Resolves #XXX` to auto-close the issue that your PR fixes (if such)

### CODE REVIEW

File: ui/web/template/channels.html [hunk 1]
```diff
@@ -34,7 +34,9 @@ <h5 class="modal-title" id="addModalLabel">Add Channel</h5>
                       </div>
                       <div class="mb-3">
                         <label for="meatadata" class="form-label">Metadata</label>
-                        <input type="text" class="form-control" name= "metadata" id="metadata">
+                        <div class="form-group">
+                          <textarea class="form-control" id="exampleFormControlTextarea1" rows="3"></textarea>
```
Comment (drasko) [inline ui/web/template/channels.html]: id must be changed, does not make sense

File: ui/api/endpoint.go [hunk 2]
```diff
@@ -151,14 +151,14 @@ func viewChannelEndpoint(svc ui.Service) endpoint.Endpoint {
 func updateChannelEndpoint(svc ui.Service) endpoint.Endpoint {
 	return func(ctx context.Context, request interface{}) (interface{}, error) {
 		req := request.(updateChannelReq)
-
+		fmt.Println("chanupd")
```
Comment (drasko) [inline ui/api/endpoint.go]: Better remove now all printfs, let's start cleaning the code.

File: ui/api/transport.go [hunk 3]
```diff
@@ -184,17 +185,15 @@ func decodeChannelsCreation(_ context.Context, r *http.Request) (interface{}, er
...
 
 func decodeChannelUpdate(_ context.Context, r *http.Request) (interface{}, error) {
-	if !strings.Contains(r.Header.Get("Content-Type"), contentType) {
-		return nil, errors.ErrUnsupportedContentType
-	}
-
+	// if !strings.Contains(r.Header.Get("Content-Type"), contentType) {
+	// 	return nil, errors.ErrUnsupportedContentType
+	// }
+	fmt.Println("paspdapsdpaspdapsdpaspdpasdpaspdpasdapsdpaspdpasd")
```
Comment (drasko) [inline ui/api/transport.go]: OK to keep validation commented for now, but remove these printfs as they polute the code

File: ui/service.go [hunk 4]
```diff
@@ -153,7 +153,7 @@ func (gs *uiService) ViewThing(ctx context.Context, token, id string) ([]byte, e
 	if err := tpl.ExecuteTemplate(&btpl, "thing", data); err != nil {
 		println(err.Error())
 	}
-	fmt.Println(btpl.String())
+	// fmt.Println(btpl.String())
```
Comment (drasko) [inline ui/service.go]: Remove

**Review by drasko — CHANGES_REQUESTED** [review]

File: ui/service.go [hunk 5]
```diff
@@ -143,51 +140,44 @@ func (gs *uiService) ViewThing(ctx context.Context, token, id string) ([]byte, e
...
-
-	return gs.ListThings(ctx, "123")
+	return gs.ViewThing(ctx, "123", id)
 }
 
 func (gs *uiService) RemoveThing(ctx context.Context, token, id string) ([]byte, error) {
-	_, err := template.ParseGlob(templateDir + "/*")
+	err := gs.sdk.DeleteThing(id, "123")
 	if err != nil {
+		fmt.Println("servisni")
```
Comment (drasko) [inline ui/service.go]: Remove printfs

File: ui/service.go [hunk 6]
```diff
@@ -143,51 +140,44 @@ func (gs *uiService) ViewThing(ctx context.Context, token, id string) ([]byte, e
...
 		return []byte{}, err
 	}
-
-	// if err := gs.thingCache.Remove(ctx, tpl); err != nil {
```
Comment (drasko) [inline ui/service.go:175]: Remove dead code

File: ui/web/template/thing.html [hunk 7]
```diff
@@ -12,12 +12,36 @@
 
           <!-- Partial goes here -->
           <div class="col-md-4">
-            <button type="button" class="btn btn-primary" data-bs-toggle="modal" data-bs-target="#addModal1">
+            <button type="button" class="btn btn-primary" data-bs-toggle="modal" data-bs-target="#updatethModal">
```
Comment (drasko) [inline ui/web/template/thing.html]: Bad naming - should be `UpdateThingModal` maybe, otherwise it is hard to read.

**Review by drasko — CHANGES_REQUESTED** [review]

File: ui/api/transport.go [hunk 8]
```diff
@@ -146,22 +152,19 @@ func decodeView(_ context.Context, r *http.Request) (interface{}, error) {
...
 	}
-
 	return req, nil
 }
 
 func decodeThingUpdate(_ context.Context, r *http.Request) (interface{}, error) {
-	if !strings.Contains(r.Header.Get("Content-Type"), contentType) {
-		return nil, errors.ErrUnsupportedContentType
-	}
+	// if !strings.Contains(r.Header.Get("Content-Type"), contentType) {
```
Comment (mteodor) [inline ui/api/transport.go]: remove comments

File: ui/service.go [hunk 9]
```diff
@@ -143,51 +139,43 @@ func (gs *uiService) ViewThing(ctx context.Context, token, id string) ([]byte, e
...
 		return []byte{}, err
 	}
-
-	return gs.ListThings(ctx, "123")
+	return gs.ViewThing(ctx, "123", id)
 }
 
 func (gs *uiService) RemoveThing(ctx context.Context, token, id string) ([]byte, error) {
-	_, err := template.ParseGlob(templateDir + "/*")
+	err := gs.sdk.DeleteThing(id, "123")
```
Comment (mteodor) [inline ui/service.go:189]: what is this `123`

Comment (fbugarski) [reply ui/service.go:189]: Single user token

File: ui/web/template/channels.html [hunk 10]
```diff
@@ -52,12 +52,12 @@ <h5 class="modal-title" id="addChannelModalLabel">Add Channel</h5>
...
                 </tr>
               </thead>
-              <tbody> 
+              <tbody>
                 {{range $i, $c := .Channels}}
                   <tr>
                     <th scope="row">{{$i}}</th>
                     <td>{{$c.Name}}</td>
-                    <td><a href={{printf "/channels/%s" $c.ID}}>{{$c.ID}}</a></td>
+                    <td><a href={{printf "/channels/%s" $c.ID}}>{{$c.ID}} </a></td>
```
Comment (drasko) [inline ui/web/template/channels.html]: Kill this space

**Review by drasko — CHANGES_REQUESTED** [review]

**Review by drasko — APPROVED** [review]
LGTM

**Pull request merged**